### ISSUE TITLE
Book Package Flow Optimization

### ISSUE THREAD

**Issue by jag3773** [unfoldingWord/book-package-app#3]
In addition to the basic reporting of what's in a BP (#2), it would be very helpful if the app could calculate the ideal next best books based on total word count of the BP.

This should accept an optional list of already completed book packages, in which case the app would identify all the parts which would have already been translated and suggest the next set of books that could be translated with the least amount of new content.

**Comment by mandolyte** [comment]
@jag3773 Definition of "Least Amount of New Content": This could be measured in at least two distinct ways:
- fewest new documents to translate
- fewest (minimum) words to translate

The definition to be used for the book package application is the second definition: fewest words.

Rationale: Consider that one document may be longer than several smaller ones. So translating the smaller ones will take less effort than the one big one. Furthermore, in some cases the smaller ones could be translated concurrently, which decreases elapsed time even more.

**Comment by mandolyte** [comment]
@jag3773 Specification for an *optional list of already completed book packages*:
**Assumptions**
This requirement assumes two languages are in play. The **from** and **to** language. In the near term, we will default/hardcode English as the **from** language (see note 1). The **to** language will have the portions that are already completed.

**Processing**
Given the user input of the **to** language, the currently translated, corresponding documents can be discovered (see notes 2 and 3 below). Once the completed work is discovered that can be used to optimize the total effort needed to completed the book book package.


**NOTES**
1. The current word count logic will not work with non-Latin scripts. See issue #https://github.com/unfoldingWord/uw-word-count/issues/4 
1. This requires the **to** language to follow the same repo and document naming conventions as English.
2. Unless releases semvers are used, we must assume that the master branch contains the documents to discover. And we must assume that existence of the translated indicates that it is actually completed.

**Comment by jag3773** [comment]
@mandolyte Automatic discovery of what is already complete is not in scope.  We may be able to add that later but right now it wouldn't be worth the extra work since we know that the assumptions required for it to be accurate aren't met in the vast majority of cases.

What *is in scope* is a way to manually check off BPs that have already been completed.

**Comment by mandolyte** [comment]
Ah, I see... let's try again...

Once the book package is presented, all the documents will have a checkbox. The user may check off which of the documents are "completed" (this could be a "what if" exercise). Once the user is done, then may regenerate the display which will be re-ordered to show flow optimization.

**Question** should the original display be ordered for flow optimization?

**Comment by jag3773** [comment]
I think that canonical order should be the default, as it will be difficult to find the book(s) you are looking for if they are in a different order. But, whatever function puts it into BPF optimized state should work without checking anything off.
